packages/url_launcher/url_launcher_platform_interface/lib/link.dart

```diff
@@ -28,35 +28,34 @@ const MethodCodec _codec = JSONMethodCodec();
 ///
 /// This is a class instead of an enum to allow future customizability e.g.
 /// opening a link in a specific iframe.
-class LinkTarget {
```

Comment:
The dart doc above explains why this is a class instead of an enum. We should either change the doc, or if the old reasoning is still valid, leave it as is.

Reply:
cc @mdebbar

Reply:
The way I understand this is that [`target` is an arbitrary string](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#target), but there are values with a special meaning, hence why this isn't an enum.

Can users add their own enum values to LinkTarget? (I can't remember if LinkTarget had a custom constructor for users though)

Reply:
LinkTarget only had a private constructor, so it wasn't user-extensible. I am happy to revert this back to a class and slap an ignore for the lint on it. Or I can remove the doc comment about the enum. Let me know what you prefer.

Reply:
I imagine this was motivated by a somewhat richer `window.open` API than what the plugin supports currently: https://developer.mozilla.org/en-US/docs/Web/API/Window/open

Reply:
The idea here was to mirror the [HTML target attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#target), which has a set of predefined values (`_blank`, `_self`, etc), but can also take an arbitrary string that refers to an already open window or iframe.

Right now we don't expose this capability to users, but if someone asks for it, we will. My preference would be to keep this as a class so when the day comes, we don't have to switch it back from enum to a class.

Reply:
Sounds good. I moved this back to be a class.